test/BoostPassVotes.t.sol

```diff
@@ -0,0 +1,260 @@
+        boostPassVotes.setTimestampAndSupply();
+
+        vm.startPrank(owner);
+        boostPassVotes.setTimestampAndSupply();
+    }
+
+    function test_get_past_votes() public {
+        // get votes before upgrade
+        uint256 previousTimepoint = boostPassVotes.boostPassCheckpoint() - 1;
+        assertEq(boostPassVotes.getPastVotes(minterAddress, previousTimepoint), 0);
+
+        mint_boost_pass();
+
+        vm.warp(block.timestamp + 1);
+
+        // get votes after upgrade (revert)
+        uint256 timepoint = boostPassVotes.clock();
+        vm.expectRevert(abi.encodeWithSelector(ERC5805FutureLookup.selector, timepoint, timepoint));
+    }
+
+    function test_get_past_total_supply() public {
+        uint256 totalSupplyBeforeMint = boostPass.totalSupply();
+
+        mint_boost_pass();
+
+        vm.warp(block.timestamp + 1);
+
+        // get supply before upgrade
+        uint256 previousTimepoint = boostPassVotes.boostPassCheckpoint() - 1;
+        assertEq(boostPassVotes.getPastTotalSupply(previousTimepoint), totalSupplyBeforeMint);
+
+        // get supply after upgrade (revert)
+        uint256 timepoint = boostPassVotes.clock();
+        vm.expectRevert(abi.encodeWithSelector(ERC5805FutureLookup.selector, timepoint, timepoint));
+        boostPassVotes.getPastTotalSupply(timepoint);
+
+        // get supply after upgrade
+        vm.warp(block.timestamp + 1);
+        assertEq(boostPassVotes.getPastTotalSupply(timepoint), boostPass.totalSupply());
+    }
+
+    function test_after_token_transfer() public {
+        vm.expectRevert(MsgSenderIsNotBoostPass.selector);
+        boostPassVotes.afterTokenTransfer(address(0), address(0));
+    }
+  
+    function test_proposal_execution() public {
+        mint_boost_pass();
+
+        // create proposal
+        bytes memory transferCallData = abi.encodeWithSignature("transfer(address,uint256)", owner, 1 ether);
+        vm.prank(owner);
+        address[] memory targets = new address[](1);
+        targets[0] = address(dai);
+        uint256[] memory values = new uint256[](1);
+        values[0] = 0;
+        bytes[] memory calldatas = new bytes[](1);
+        calldatas[0] = transferCallData;
+        string memory description = "Proposal #1: Transfer DAI to the owner";
+        uint256 proposalId = myGovernor.propose(targets, values, calldatas, description);
+
+        // Cast votes
+        uint256 snapshot = myGovernor.proposalSnapshot(proposalId);
+        vm.warp(snapshot + 1);
+        vm.prank(minterAddress);
+        myGovernor.castVote(proposalId, 1);
+
+        // Execute proposal
+        uint256 deadline = myGovernor.proposalDeadline(proposalId);
+        vm.warp(deadline + 1);
+        uint256 successfulProposalId = myGovernor.execute(targets, values, calldatas, keccak256(abi.encodePacked(description)));
+
+        assertEq(successfulProposalId, proposalId);
+        assertEq(dai.balanceOf(owner), 1 ether);
+    }
+
+    function test_delegate() public {
```

Comment:
I'd like to see some multi-step delegation tests, what it looks like when someone "undelegates" and some testing around what happens when address(0) is passed into delegate. Ideally some fuzz testing where we iterate through a stack of delegators but that may be too heavy here. Regardless undelegated and multi-step delegation would be good to test.